OutOfSchool/OutOfSchool.WebApi/Services/ChildService.cs

```diff
@@ -343,15 +367,37 @@ public async Task DeleteChildCheckingItsUserIdProperty(Guid id, string userId)
         if (!string.IsNullOrWhiteSpace(filter.SearchString))
         {
             var tempPredicate = PredicateBuilder.False<Child>();
-
-            foreach (var word in filter.SearchString.Split(' ', ',', StringSplitOptions.RemoveEmptyEntries))
+            if (filter.SearchString.Length >= 3)
             {
-                tempPredicate = tempPredicate.Or(
-                    x => x.FirstName.StartsWith(word, StringComparison.InvariantCultureIgnoreCase)
-                        || x.LastName.StartsWith(word, StringComparison.InvariantCultureIgnoreCase)
-                        || x.MiddleName.StartsWith(word, StringComparison.InvariantCultureIgnoreCase)
-                        || x.Parent.User.Email.StartsWith(word, StringComparison.InvariantCultureIgnoreCase)
-                        || x.Parent.User.PhoneNumber.Contains(word, StringComparison.InvariantCulture));
+                filter.SearchString = filter.SearchString.Trim();
+                string phoneNumber = "";
+                if (filter.SearchString[0] == '+')
+                {
+                    for (int i = 0; i < filter.SearchString.Length - 1; i++)
+                    {
+                        phoneNumber += $"{filter.SearchString[i + 1]}";
+                    }
+                }
+
+                if (phoneNumber.All(t => char.IsNumber(t)))
+                {
+                    foreach (var word in filter.SearchString.Split(' ', ',', StringSplitOptions.RemoveEmptyEntries))
+                    {
+                        tempPredicate = tempPredicate.Or(
+                            x => x.Parent.User.PhoneNumber.Contains(word, StringComparison.InvariantCulture));
+                    }
+                }
+                else
+                {
+                    foreach (var word in filter.SearchString.Split(' ', ',', StringSplitOptions.RemoveEmptyEntries))
+                    {
+                        tempPredicate = tempPredicate.Or(
+                            x => x.FirstName.StartsWith(word, StringComparison.InvariantCultureIgnoreCase)
+                                 || x.LastName.StartsWith(word, StringComparison.InvariantCultureIgnoreCase)
+                                 || x.MiddleName.StartsWith(word, StringComparison.InvariantCultureIgnoreCase)
+                                 || x.Parent.User.Email.StartsWith(word, StringComparison.InvariantCultureIgnoreCase));
+                    } 
+                }
```

Comment:
Smthn like this:
foreach (var word in filter.SearchString.Split(' ', ',', StringSplitOptions.RemoveEmptyEntries))
{
	If (word.Any(c => char.IsLetter(c))){
		tempPredicate = tempPredicate.Or(
			x => x.FirstName.StartsWith(word, StringComparison.InvariantCultureIgnoreCase)
				 || x.LastName.StartsWith(word, StringComparison.InvariantCultureIgnoreCase)
				 || x.MiddleName.StartsWith(word, StringComparison.InvariantCultureIgnoreCase)
				 || x.Parent.User.Email.StartsWith(word, StringComparison.InvariantCultureIgnoreCase));	
	}
	else{
		string phoneNumber = word.Where(c => char.IsNumber(c));
		
		tempPredicate = tempPredicate.Or(
			x => x.Parent.User.PhoneNumber.Contains(phoneNumber, StringComparison.InvariantCulture));
	}
}